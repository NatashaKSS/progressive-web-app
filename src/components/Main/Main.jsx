import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Home from '../Home/Home.jsx';
import About from '../About/About.jsx';
import PostCollection from '../PostCollection/PostCollection.jsx';

import styles from './Main.css';

const Main = ({children}) => {
  return (
    <main>
      <Switch>
        <Route exact path='/' component={Home}/>
        <Route exact path='/about' component={About}/>
        <Route path='/posts' component={PostCollection}/>
      </Switch>
    </main>
  );
}

export default Main;
