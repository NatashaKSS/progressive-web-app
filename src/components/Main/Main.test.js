import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Main from './Main';

test('<Main /> should render as expected', () => {
  const component = shallow(<Main />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
