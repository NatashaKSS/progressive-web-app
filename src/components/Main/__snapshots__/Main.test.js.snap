// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render as expected 1`] = `
<main>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/about"
    />
    <Route
      component={[Function]}
      path="/posts"
    />
  </Switch>
</main>
`;
